backend/src/error.rs

```diff
@@ -0,0 +1,64 @@
+#![allow(clippy::module_name_repetitions)]
+use std::fmt;
+
+use axum::response::{IntoResponse, Response};
+use error_stack::Context;
+use hyper::StatusCode;
+use thiserror::Error;
+
+/// Generic error that wraps `error_stack::Context`.
+/// Generally used for notifying the client that some error occurred on the server
+///
+/// For the internal server errors
+#[derive(Debug)]
+pub struct InternalAppError;
```

Comment:
I think, there is no need to declare every error as a separate struct. For small/medium size projects (like this one) it is simpler to declare one error type as `enum` and use it everywhere.

Example:
```
#[derive(Error, Debug)]
pub enum ManGuiErrorKind {
    #[error(...)]
    InternalAppError,
    ...
}

pub type ManGuiError = Report<ManGuiErrorKind>;
pub type ManGuiResult<T> = Result<T, ManGuiError>;
```